token/js/src/extensions/groupPointer/instructions.ts

```diff
@@ -0,0 +1,98 @@
+import { struct, u8 } from '@solana/buffer-layout';
+import { publicKey } from '@solana/buffer-layout-utils';
+import type { Signer } from '@solana/web3.js';
+import { PublicKey, TransactionInstruction } from '@solana/web3.js';
+import { TOKEN_2022_PROGRAM_ID, programSupportsExtensions } from '../../constants.js';
+import { TokenUnsupportedInstructionError } from '../../errors.js';
+import { TokenInstruction } from '../../instructions/types.js';
+import { addSigners } from '../../instructions/internal.js';
+
+export enum GroupPointerInstruction {
+    Initialize = 0,
+    Update = 1,
+}
+
+export const initializeGroupPointerData = struct<{
+    instruction: TokenInstruction.GroupPointerExtension;
+    groupPointerInstruction: number;
+    authority: PublicKey;
+    groupAddress: PublicKey;
+}>([
+    // prettier-ignore
+    u8('instruction'),
+    u8('groupPointerInstruction'),
+    publicKey('authority'),
+    publicKey('groupAddress'),
+]);
+
+/**
+ * Construct an Initialize GroupPointer instruction
+ *
+ * @param mint            Token mint account
+ * @param authority       Optional Authority that can set the group address
+ * @param groupAddress    Optional Account address that holds the group
+ * @param programId       SPL Token program account
+ *
+ * @return Instruction to add to a transaction
+ */
+export function createInitializeGroupPointerInstruction(
+    mint: PublicKey,
+    authority: PublicKey | null,
+    groupAddress: PublicKey | null,
+    programId: PublicKey
```

Comment:
nit for consistency with the other ix:
```suggestion
    programId: PublicKey = TOKEN_2022_PROGRAM_ID
```